spec/workers/detect_invariants_daily_check_spec.rb

```diff
@@ -141,6 +141,7 @@
       create(:completed_application_form, submitted_application_choices_count: 3)
       bad_application_form = create(:completed_application_form, submitted_application_choices_count: 2)
       bad_application_form.application_choices << build_list(:submitted_application_choice, 2, status: :offer)
+      ApplicationChoice.all.each { |a| a.update_course_option_and_associated_fields! create(:course_option) }
```

Comment:
`:completed_application_form` factory doesn't seem to trigger `after` hooks for application choices. This leaves `provider_ids` empty and the provider_ids check triggers an additional Sentry.

Reply:
ah interesting!  might be worth updating the application choice factories to ensure this

Reply:
This could be a significant refactor I would rather avoid. The issue is probably related to the fact that the application choice factories are called indirectly, as we are using application form factories here.